torch/distributed/_composable/fsdp/_fsdp_param_group.py

```diff
@@ -220,3 +305,20 @@ def _get_param_module_infos(
     if len(param_to_module_info) != len(params):
         raise AssertionError(f"Some parameters are not in the module tree of {module}")
     return [param_to_module_info[param] for param in params]
+
+
+class RegisterPostBackwardHook(torch.autograd.Function):
```

Comment:
this part feels mysterious to me. is there some api doc we can refer to regarding inheriting ``torch.autograd.Function`` and call ``.apply(tensor)`` ?

Reply:
https://pytorch.org/docs/stable/notes/extending.html#extending-autograd
https://pytorch.org/tutorials/beginner/examples_autograd/two_layer_net_custom_function.html